decidim-proposals/app/controllers/decidim/proposals/admin/participatory_texts_controller.rb

```diff
@@ -75,6 +77,18 @@ def update
             end
           end
         end
+
+        # Removes all the unpublished proposals (drafts).
+        def discard
+          enforce_permission_to :manage, :participatory_texts
+
+          DiscardParticipatoryText.call(current_component) do
+            on(:ok) do
+              flash[:notice] = I18n.t("participatory_texts.discard.success", scope: "decidim.proposals.admin")
+              redirect_to participatory_texts_path(component_id: current_component.id, initiative_slug: "asdf")
```

Comment:
Sorry, what do you mean?
If you are referring to the initiative_slug it is there to make the tests work, without this parameter they crash even when there is no initiative involved. We were dedicating time with @mrcasals to investigate and did not found the cause. As it is not affecting running code we agreed to leave the parameter and get the tests work.

Reply:
I mean the `initiative_slug: "asdf"` part

Reply:
What do you mean it's not affecting running code?

Reply:
I meant that adding this parameter does not affect the resulting path, but still tests require this parameter to be present, even when we're not in an initiative component.

Reply:
Which test does require this parameter? I think we should look further into this, participatory texts shouldn't need anything from initiatives, right?

Reply:
And even if users are redirected to the same page, having `initiative_slug=asdf` as a parameter at a production URL is super weird to me.

Reply:
@oriolgual you can for example remove the 3 `initiative_slug=asdf` from `participatory_texts_controller.rb` and run `rspec ./spec/controllers/decidim/admin/participatory_texts_controller_spec.rb`. This will result in
```
     Failure/Error: redirect_to participatory_texts_path(component_id: current_component.id)
     
     ActionController::UrlGenerationError:
       No route matches {:component_id=>865}, missing required keys: [:initiative_slug]
```

Reply:
I've made some changes to avoid doing this. I'm still not sure why there are problems in this particular controller, but my guess is because participatory texts aren't a resourceable.